packages/kbn-unified-data-table/src/components/data_table_additional_display_settings.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { EuiFormRow, EuiRange } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { debounce } from 'lodash';
+
+export const MAX_ALLOWED_SAMPLE_SIZE = 1000;
```

Comment:
Shouldn't we set to 500, which is the default in Advanced settings, and limit the max to `discover:sampleSize`? We now have the option to load more anyway. I've configured 2000 for this setting, still was was 1000. So I think taking discover:sampleSize into consideration makes sense, also it should be used e.g. when an admin decides to go back from 1000 to 500 , the user configured 990, it could make sense to limit the request to 500 in this case, wdyt?
This is what came to my mind while testing it. Apart from that: 🎉

Reply:
@kertal That's a good idea! The only issues is that if we base the max size on `discover:sampleSize` then we can't set it for the schema validation. Also users can even now update it in the URL and request more than the configured `discover:sampleSize`, would you limit that too?

Reply:
We could define 10_000 limit for the schema and in UI use min of 10_000 and `discover:sampleSize`.

Reply:
A hard limit of 10000 could be a good idea for the schema ... however we should make sure the number requested from ES should be limited by the sampleSize. Because in the end, this it was matters in terms of traffic and performance.

Reply:
@kertal I pushed changes for it https://github.com/elastic/kibana/pull/157269/commits/976661552c5ef4a73a6ebccac8a1473edbddba7c

Could you please check if it's close to what you meant?

Reply:
i will, thx!

Reply:
Tested and it's exactly what I meant! Thx 🙏 !